### PR TITLE
Enabling Windows build

### PR DESCRIPTION
**Pull request by himanshupathak21061998**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

I am trying to enable windows build windows build for ensmallen and complete my whole task to enable build for mlpack.

### CODE REVIEW

**Comment by himanshupathak21061998** [comment]
@conda-forge-manager , please rerender

**Comment by himanshupathak21061998** [comment]
@conda-forge-admin , please rerender

**Comment by himanshupathak21061998** [comment]
@conda-forge-admin , please rerender

**Comment by himanshupathak21061998** [comment]
@conda-forge-admin , please rerender

**Comment by himanshupathak21061998** [comment]
@conda-forge-admin , please rerender

**Comment by himanshupathak21061998** [comment]
Hey @marcelotrevisani I was trying to add vc package for windows but I am getting this error
```
conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform win-64: {"vc[version='14.*,9,>=14,<15.0a0']"}

```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,8 +22,11 @@ requirements:
   host:
     - armadillo >=6.500.0
     - libblas
-    - libcblas
+    - libcblas  # [not win]
+    - libcblas ==3.8  # [win]
     - openmp
+    - vc ==14.*  # [win]
+    - vc ==9  # [win]
```
Comment (himanshupathak21061998) [inline recipe/meta.yaml]: As you can see here I added version of vc which are available. But getting error can you please explain.

**Comment by marcelotrevisani** [comment]
@conda-forge-admin , please rerender

**Comment by marcelotrevisani** [comment]
@conda-forge-admin , please rerender

**Comment by marcelotrevisani** [comment]
Unfortunately, I am a bit busy these days and I will not have time to take a look at that a bit further.
But the compilation is failing, it cannot build on vc 18 as well, so I just skipped.

Maybe @rcurtin has an idea here

The error is because some kind of template deduction, maybe is a bug on msvc, I don't know, I didn't have time to investigate

```
d:\bld\ensmallen_1579674194756\work\include\ensmallen_bits\problems\aug_lagrangian_test_functions_impl.hpp(139): error C2672: 'std::min': no matching overloaded function found
d:\bld\ensmallen_1579674194756\work\include\ensmallen_bits\aug_lagrangian\aug_lagrangian_impl.hpp(114): note: see reference to function template instantiation 'float ens::test::GockenbachFunction::EvaluateConstraint<BaseMatType>(const std::size_t,const MatType &)' being compiled
        with
        [
            MatType=BaseMatType
        ]
d:\bld\ensmallen_1579674194756\work\include\ensmallen_bits/aug_lagrangian/aug_lagrangian.hpp(205): note: see reference to function template instantiation 'bool ens::AugLagrangian::Optimize<LagrangianFunctionType,MatType,MatType,>(ens::AugLagrangianFunction<LagrangianFunctionType> &,MatType &)' being compiled
        with
        [
            LagrangianFunctionType=ens::test::GockenbachFunction,
            MatType=arma::fmat
        ]
d:\bld\ensmallen_1579674194756\work\include\ensmallen_bits\problems\aug_lagrangian_test_functions_impl.hpp(139): error C2780: '_Ty std::min(std::initializer_list<_Elem>)': expects 1 arguments - 2 provided
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\algorithm(3654): note: see declaration of 'std::min'
d:\bld\ensmallen_1579674194756\work\include\ensmallen_bits\problems\aug_lagrangian_test_functions_impl.hpp(139): error C2782: 'const _Ty &std::min(const _Ty &,const _Ty &) noexcept(<expr>)': template parameter '_Ty' is ambiguous
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\algorithm(3646): note: see declaration of 'std::min'
d:\bld\ensmallen_1579674194756\work\include\ensmallen_bits\problems\aug_lagrangian_test_functions_impl.hpp(139): note: could be 'float'
d:\bld\ensmallen_1579674194756\work\include\ensmallen_bits\problems\aug_lagrangian_test_functions_impl.hpp(139): note: or       'double'
d:\bld\ensmallen_1579674194756\work\include\ensmallen_bits\problems\aug_lagrangian_test_functions_impl.hpp(139): error C2784: 'const _Ty &std::min(const _Ty &,const _Ty &) noexcept(<expr>)': could not deduce template argument for 'const _Ty &' from 'float'
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\algorithm(3646): note: see declaration of 'std::min'
d:\bld\ensmallen_1579674194756\work\include\ensmallen_bits\problems\aug_lagrangian_test_functions_impl.hpp(139): error C2784: '_Ty std::min(std::initializer_list<_Elem>,_Pr)': could not deduce template argument for 'std::initializer_list<_Elem>' from 'double'
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\algorithm(3635): note: see declaration of 'std::min'
d:\bld\ensmallen_1579674194756\work\include\ensmallen_bits\problems\aug_lagrangian_test_functions_impl.hpp(139): error C2780: 'const _Ty &std::min(const _Ty &,const _Ty &,_Pr) noexcept(<expr>)': expects 3 arguments - 2 provided
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\algorithm(3625): note: see declaration of 'std::min'

```

**Comment by marcelotrevisani** [comment]
I just found this on Microsoft documentation, that explains those errors

> Versions of the compiler before Visual Studio 2017 did not perform access checking on qualified names in some template contexts. This can interfere with expected SFINAE behavior where the substitution is expected to fail due to the inaccessibility of a name. This could have potentially caused a crash or unexpected behavior at runtime due to the compiler incorrectly calling the wrong overload of the operator. In Visual Studio 2017, a compiler error is raised.

**Comment by marcelotrevisani** [comment]
@conda-forge-admin , please rerender

**Comment by rcurtin** [comment]
@marcelotrevisani glad you got it working. :+1: I encountered this issue also while getting AppVeyor working for ensmallen: https://github.com/mlpack/ensmallen/pull/158.  So, it should be fixed in the next release.

**Comment by himanshupathak21061998** [comment]
Thanks @marcelotrevisani for handling this.

**Pull request merged**